waspc/cli/src/Wasp/Cli/Command/CreateNewProject/Templates.hs

```diff
@@ -0,0 +1,79 @@
+module Wasp.Cli.Command.CreateNewProject.Templates
+  ( getStarterTemplates,
+    StarterTemplateNames,
+    templatesToList,
+    isValidTemplateName,
+  )
+where
+
+import Control.Exception (try)
+import Data.Aeson (FromJSON, parseJSON, withObject, (.:))
+import qualified Network.HTTP.Simple as HTTP
+
+data StarterTemplateNames = StarterTemplateNames [String]
+
+getStarterTemplates :: IO (Maybe StarterTemplateNames)
+getStarterTemplates = do
+  -- Github returns 403 if we don't specify user-agent.
+  let request = HTTP.addRequestHeader "User-Agent" "wasp-lang/wasp" templatesRepoInfoURL
+  responseOrException <- try $ HTTP.httpJSONEither request
+  return $ case responseOrException of
+    Right response -> extractTemplateNames . HTTP.getResponseBody $ response
+    Left (_ :: HTTP.HttpException) -> Nothing
+  where
+    templatesRepoInfoURL :: HTTP.Request
+    templatesRepoInfoURL = "https://api.github.com/repos/wasp-lang/starters/git/trees/main"
+
+    -- Each folder in the "wasp-lang/starters" repo is a template.
+    extractTemplateNames :: Either HTTP.JSONException RepoInfo -> Maybe StarterTemplateNames
+    extractTemplateNames (Left _) = Nothing
+    extractTemplateNames (Right body) = Just . StarterTemplateNames . map _path . filter isFolder $ _objects body
```

Comment:
Weird that it takes Either and then returns Maybe where Left and Nothing are directly mapped. Then you probably just want a function that takes normal value and returns normal value and then the caller should handle case when there are no input values.

Reply:
> Then you probably just want a function that takes normal value and returns normal value and then the caller should handle case when there are no input values.

I'm not sure what you mean here exactly, could you give me an example?